tests/test_sailor/test_sap_iot/test_fetch.py

```diff
@@ -284,9 +284,10 @@ def test_get_indicator_data_timeout_reached(self, mock_check, caplog, mock_zipfi
         mock_gzip.side_effect = [BytesIO(make_csv_bytes(1, ''))]
         mock_check.return_value = False
 
-        with pytest.raises(TimeoutError, match='Timeout of 2 seconds was reached for fetching indicator data.'):
-            get_indicator_data('2020-01-01T00:00:00Z', '2020-02-01T00:00:00Z', indicator_set, equipment_set,
-                               timeout="P2S")
+        with pytest.raises(TimeoutError, match='Timeout of 0 seconds was reached for fetching indicator data.'):
+            with patch('time.sleep'):
+                get_indicator_data('2020-01-01T00:00:00Z', '2020-02-01T00:00:00Z', indicator_set, equipment_set,
+                                   timeout="P0S")
```

Comment:
There's no problem. In some subsequent iteration of the loop, the term would be greater than 0. Looking at the mocks, we can see that the test can sustain multiple iterations of the loop.

However, since you pointed me to look at this once more, I found some opportunity to delete some unnecessary test code.

Reply:
Ok, that makes sense.